### PR TITLE
feat: Updates `WalletConnect` to `v2`

### PR DESCRIPTION
**Pull request by JoaquinBattilana**
## General Changes

- Added a new `WalletConnect` connector for using the new version of `WalletConnect` sdk

## Developer

- There is a bug with the `enable` method of the `WalletConnect` provider, it returns a `Promise` but the promise only resolve when the user connects correctly, when the user closes the modal the `Promise` is still hanging.

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [ ]  The base branch is set to `main`
- [ ]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [ ]  The Github issue has been linked to the PR in the Development section
- [ ]  The General Changes section has been filled out
- [ ]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [ ]  The PR is in `Open` state and not in `Draft` mode
- [ ]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -2,4 +2,5 @@ NEXT_PUBLIC_ENV=prod
 NEXT_PUBLIC_API_BASEURL=https://aave-api-v2.aave.com
 #NEXT_PUBLIC_FORK_BASE_CHAIN_ID=1
 #NEXT_PUBLIC_FORK_CHAIN_ID=1
-#NEXT_PUBLIC_FORK_URL_RPC="https://rpc.tenderly.co/fork/1234567"
\ No newline at end of file
+#NEXT_PUBLIC_FORK_URL_RPC="https://rpc.tenderly.co/fork/1234567"
+NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID=
```
Comment (JoaquinBattilana) [inline .env.development:6]: We need a wallet connect project id @foodaka @0x4Graham

**Comment by MareskoY** [comment]
it's bugged 

if try to connect via walletConnect we get error

Screenshot 2023-02-10 at 14 11 34

**Comment by 0x4Graham** [comment]
With what wallet?

From: Nikita ***@***.***>
Date: Friday, 10 February 2023 at 15:12
To: aave/interface ***@***.***>
Cc: 0xGraham ***@***.***>, Mention ***@***.***>
Subject: Re: [aave/interface] Draft: feat: Updates `WalletConnect` to `v2` (PR #1470)

it's bugged

if try to connect via walletConnect we get error

[Screenshot 2023-02-10 at 14 11 34]<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F18032275%2F218112964-3579819b-3b5a-4209-a7b4-c7ca085c9b41.png&data=05%7C01%7C%7C6d614e35a8234ce5d27d08db0b70e771%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638116351778968351%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=KqjJ8ku6cSdXaMJCht8S%2BqRHQ76BXv%2FvLqMzjwPH9nE%3D&reserved=0>

**Comment by JoaquinBattilana** [comment]
Oh yes, this PR is tagged as `Draft` because:
1) we're waiting for the creation of the `projectId`
2) we're waiting for `WalletConnect` team to address a bug with the modal state.

So it wont work for now.

**Comment by grothem** [comment]
Link T-5133

**Review by defispartan — DISMISSED** [review]

**Review by 0x4Graham — DISMISSED** [review]

**Review by 0x4Graham — APPROVED** [review]

**Review by grothem — APPROVED** [review]

**Pull request merged**